networkd/network_monitor_thread.ml

```diff
@@ -113,10 +107,11 @@ let get_link_stats () =
 		List.map (fun link ->
 			(standardise_name (Link.get_name link)), link
 		) links |>
+		(* Only keep interfaces with prefixes on the whitelist, and exclude VLAN
+		   devices (ethx.y). *)
 		List.filter (fun (name,link) ->
-			let is_monitor_blacklisted = List.exists (fun s -> String.startswith s name) !monitor_blacklist ||
-						(String.startswith "eth" name && String.contains name '.') in
-			not is_monitor_blacklisted
+			List.exists (fun s -> String.startswith s name) !monitor_whitelist
```

Comment:
It is simple enough but I would find creating two predicates even clearer: `is_whitelisted` and `is_vlan` and then use `List.filter (fun (name,_) -> is_whitelisted name && not (is_vlan name)`.